src/sql/workbench/browser/modelComponents/treeViewDataProvider.ts

```diff
@@ -32,4 +49,29 @@ export class TreeViewDataProvider extends vsTreeView.TreeViewDataProvider implem
 
 	refresh(itemsToRefreshByHandle: { [treeItemHandle: string]: ITreeComponentItem }) {
 	}
+
+	getChildren(treeItem?: ITreeComponentItem): Promise<ITreeComponentItem[]> {
+		return Promise.resolve(this._proxy.$getChildren(this.treeViewId, treeItem ? treeItem.handle : undefined)
+			.then(
+				children => this.treeComponentPostGetChildren(children),
+				err => {
+					this.notificationService.error(err);
+					return [];
+				}));
+	}
+
+	private async treeComponentPostGetChildren(elements: ITreeComponentItem[]): Promise<ResolvableTreeComponentItem[]> {
```

Comment:
```
	protected async postGetChildren(elements: ITreeComponentItem[]): Promise<ResolvableTreeComponentItem[]> {
```

Then make the postGetChildren() protected in the base class too

Reply:
I'd also add a //override comment on top of the methods you're overriding to make it obvious why thery're there